test/ui-test/testSuites/suite_messaging/tst_ChatFlow/test.feature

```diff
@@ -1,169 +1,173 @@
-
-	@merge
-    Scenario: User can edit a message
-         When user joins chat room test
-         Then user is able to send chat message
-         | message               |
-         | Edit me                 |
-         When the user edits the message at index 0 and changes it to "Edited by me"
-         Then the message (edited) is displayed in the last message
-
-
-    @mayfail @merge
-    Scenario: User can reply to another user's message
-         When user joins chat room test
-         Then the user can reply to the message at index 0 with "This is a reply to another user"
-
-	@mayfail @merge
-    Scenario: User joins a room and marks it as read
-		 When user joins chat room test
-		 Then the user can mark the channel test as read
-         # TODO find a way to validate that it worked
-
-	@merge
-    Scenario: User can delete their own message
-         When user joins chat room automation-test
-         Then the user is able to send a random chat message
-         Then the user can delete the message at index 0
-         Then the last message is not the random message
-
-	@merge
-    Scenario: User can clear chat history
-        When user joins chat room test
-        Then user is able to send chat message
-        | message             |
-        | Hello               |
-        | How are you         |
-        | I am from status    |
-        | tell me how you do? |
-        When the user clears chat history
+    # TODO: Scenario: The user can reply to their OWN message
+    @mayfail
+    # TODO: It works standalone but when it runs as part of the sequence, the action of reply is not done always. The popup option does not appear.
+    Scenario Outline: The user can reply to the last message
+         When the user sends a chat message <message>
+         And the user replies to the message at index 0 with <reply>
+         # TODO: Check the last message is really a reply, now just checking the last message is the expected one but could not be a reply. The popup option does not appear.
+         Then the chat message <reply> is displayed as a reply
+         Examples:
+		 | message			   | reply			 |
+		 | random chat message | This is a reply |
+
+	@mayfail
+	# TODO: It works standalone but when it runs as part of the sequence, the action of edit is not done always. The popup option does not appear.
+    Scenario Outline: The user can edit a message
+         When the user sends a chat message <message>
+         And the user edits the message at index 0 and changes it to <edited>
+         # TODO: Check last message is an edited one, now just checking the last message is the expected one but could not be an edited one.
+         Then the chat message <edited> is displayed as an edited one
+         Examples:
+		 | message | edited		  |
+		 | Edit me | Edited by me |
+
+    @relyon-mailserver
+    Scenario Outline: The user can reply to another message
+         When the user replies to the message at index 0 with <reply>
```

Comment:
I see this statement as the actions not as a precondition. So I should keep the `when` here. WDYT?

Reply:
I guess it's a matter of preferred style - personally I find `Given` improves readability by providing context but leaving out `When` does also leave the action as only implied instead of written. If you prefer `When` that's fine to keep it.  

Also, if you agree that all scenarios should explicitly state the action then we can say that all scenarios should have at least a `when` and `then` statement.

Reply:
Yes, I agree on that. For me it makes sense to have all scenarios, at least, with an action (when) and a verification (then), and depending on the contexts, it can need also a precondition (given).